src/panels/lovelace/editor/hui-entity-row-editor.ts

```diff
@@ -0,0 +1,343 @@
+import { safeDump, safeLoad } from "js-yaml";
+import {
+  css,
+  CSSResult,
+  customElement,
+  internalProperty,
+  LitElement,
+  property,
+  query,
+} from "lit-element";
+import { html, TemplateResult } from "lit-html";
+import { fireEvent } from "../../../common/dom/fire_event";
+import { computeRTL } from "../../../common/util/compute_rtl";
+import { deepEqual } from "../../../common/util/deep-equal";
+import "../../../components/ha-circular-progress";
+import "../../../components/ha-code-editor";
+import type { HaCodeEditor } from "../../../components/ha-code-editor";
+import { HomeAssistant } from "../../../types";
+import { handleStructError } from "../common/structs/handle-errors";
+import {
+  DOMAIN_TO_ELEMENT_TYPE,
+  getRowElementClass,
+} from "../create-element/create-row-element";
+import { LovelaceRowConfig } from "../entity-rows/types";
+import { LovelaceRowEditor } from "../types";
+import { GUISupportError } from "./gui-support-error";
+
+export interface RowConfigChangedEvent {
+  config: LovelaceRowConfig;
+  error?: string;
+  guiModeAvailable?: boolean;
+}
+
+declare global {
+  interface HASSDomEvents {
+    "row-config-changed": RowConfigChangedEvent;
+  }
+}
+
+export interface UIRowConfigChangedEvent extends Event {
+  detail: {
+    config: LovelaceRowConfig;
+  };
+}
+
+@customElement("hui-entity-row-editor")
```

Comment:
This entire file feels like a copy of the card editor?

Reply:
100%

Reply:
Few changes to make it be for the row

Reply:
Can we generalize the card editor, so we have less duplicate code?

Reply:
I cant try. I think the biggest change is the way to find the correct type. But should be pretty easy